src/DotNet.Status.Web/Controllers/AzurePipelinesController.cs

```diff
@@ -0,0 +1,375 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.DotNet.AzureDevOpsTimeline;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Octokit;
+using Issue = Octokit.Issue;
+
+namespace DotNet.Status.Web.Controllers
+{
+    [ApiController]
+    [Route("api/azp")]
+    public class AzurePipelinesController : ControllerBase
+    {
+        private readonly GitHubClientFactory _gitHubClientFactory;
+        private readonly IOptions<BuildMonitorOptions> _options;
+        private readonly ILogger<AzurePipelinesController> _logger;
+        private readonly Lazy<AzureDevOpsClient> _clientLazy;
+        private readonly Lazy<Task<Dictionary<string, string>>> _projectMapping;
+
+        public AzurePipelinesController(
+            GitHubClientFactory gitHubClientFactory,
+            IOptions<BuildMonitorOptions> options,
+            ILogger<AzurePipelinesController> logger)
+        {
+            _gitHubClientFactory = gitHubClientFactory;
+            _options = options;
+            _logger = logger;
+            _clientLazy = new Lazy<AzureDevOpsClient>(BuildAzureDevOpsClient);
+            _projectMapping = new Lazy<Task<Dictionary<string,string>>>(GetProjectMappingInternal);
+        }
+
+        private AzureDevOpsClient Client => _clientLazy.Value;
+
+        private AzureDevOpsClient BuildAzureDevOpsClient()
+        {
+            AzurePipelinesMonitorOptions o = _options.Value.Monitor;
+            return new AzureDevOpsClient(o.BaseUrl, o.Organization, o.MaxParallelRequests, o.AccessToken);
+        }
+
+        private async Task<Dictionary<string, string>> GetProjectMappingInternal()
+        {
+            var projects = await Client.ListProjects();
+            return projects.ToDictionary(p => p.Id, p => p.Name);
+        }
+
+        private async Task<string> GetProjectNameAsync(string id)
+        {
+            Dictionary<string, string> mapping = await _projectMapping.Value;
+            return mapping.GetValueOrDefault(id);
+        }
+
+        [HttpPost]
+        [Route("build-complete")]
+        [ProducesResponseType((int) HttpStatusCode.NoContent)]
+        public async Task<IActionResult> BuildComplete(AzureDevOpsEvent<AzureDevOpsMinimalBuildResource> buildEvent)
+        {
+            _logger.LogInformation("Build complete notification for build '{buildId}', URL: {buildUrl}",
```

Comment:
Perhaps adding repo+branch to the message can be useful in the future for filtering the logs.

Reply:
We don't have it.  The "buildEvent" literally contains only the build ID.

We don't get the actual branch information until _waaaaay_ later, around line 82.  I can add a little log there, but we are going to get a _lot_ of builds here (literally every build on public and internal), so I wanted to keep the logging minimal until we decided to actually take action on a build (then the logging gets a lot more verbose)